src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -174,24 +163,23 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                     getCertificateRequest.getCertificateRequestOptions().getCertificateType();
             // TODO: Should be configurable
             KeyPair keyPair = CertificateStore.newRSAKeyPair(4096);
-            X509Certificate[] caCertificates = getX509CACertificates();
 
             if (certificateType.equals(GetCertificateRequestOptions.CertificateType.SERVER)) {
-                Consumer<X509Certificate> consumer = (t) -> {
+                BiConsumer<X509Certificate, X509Certificate[]> consumer = (serverCert, caCertificates) -> {
                     CertificateUpdateEvent certificateUpdateEvent =
-                            new CertificateUpdateEvent(keyPair, t, caCertificates);
+                            new CertificateUpdateEvent(keyPair, serverCert, caCertificates);
```

Comment:
I think we still have the same problem here, except with the KeyPair. But I think we can solve that problem later.

Reply:
We are good for the keyPair not changing. From that key pair we only use the public key to generate the cert although it would be nice if we re generated a new one